### PR TITLE
Various output routine updates

### PR DESCRIPTION
**Pull request by pzinn**
- moved some TeX-related code from `expressions.m2` to `latex.m2`
- fixed an annoying double superscript bug for things like `x'_1^2`
- harmonised tex output of zero vectors
- added `ReverseDictionary` to iterators
- added `texMath RRi`

### CODE REVIEW

File: M2/Macaulay2/m2/latex.m2 [hunk 1]
```diff
@@ -72,20 +79,84 @@ sectionType = sectionNumber -> (
     "\\subsubsubsubsubparagraph");
 
 -----------------------------------------------------------------------------
+keywordTexMath = applyKeys(hashTable { -- both unary and binary keywords
+    symbol |- => "\\vdash ",
```
Comment (mahrud) [inline M2/Macaulay2/m2/latex.m2:83]: Two comments:
1. The indenting is incorrect with the `applyKeys(..., symbolBody)` addition. Could you do that on the next line perhaps? (e.g. rewrite it with `keywordTexMath = applyKeys(keywordTexMath, symbolBody)`)
2. On that note, could you also adjust the entries so the `=>` are all aligned?

File: M2/Macaulay2/m2/latex.m2 [hunk 2]
```diff
@@ -72,20 +79,84 @@ sectionType = sectionNumber -> (
...
+},last)
```
Comment (mahrud) [inline M2/Macaulay2/m2/latex.m2:140]: Could you break these into multiple lines? (Also I think the indentation is off)

File: M2/Macaulay2/m2/latex.m2 (hunk 1 above)
Comment (pzinn) [reply M2/Macaulay2/m2/latex.m2:83]: what do you mean, the indenting is incorrect? do we have rules for indentation laid out somewhere?

**Comment by pzinn** [comment]
those inter-dependencies of various files in the loadsequence are a pain

**Comment by pzinn** [comment]
so with this PR: `iterators.m2` needs `expressions.m2` which needs `max.m2` which needs `fold.m2` which needs `iterators.m2`...
how are we supposed to deal with those circular dependencies?

**Comment by d-torrance** [comment]
I suppose the `accumulate` and `fold` methods involving `Thing` (i.e., the ones that use iterators) could get moved from `fold.m2` to `iterators.m2` to remove that dependency.

File: M2/Macaulay2/m2/latex.m2 (hunk 1 above)
Comment (mahrud) [reply M2/Macaulay2/m2/latex.m2:83]: [Yes](https://github.com/Macaulay2/M2-emacs/blob/90ce16a15ea1f4fad08e8ba0b74b5045f4a72e8f/M2.el#L39-L42), four spaces for each unmatched parenthesis, etc. (and ideally tabs for each eight spaces, but this is more aesthetic).

**Comment by DanGrayson** [comment]
Could you fix the broken test-builds, please?

**Comment by pzinn** [comment]
a `NumericalImplicitization` test runs oom, don't think it's related to this PR?

**Comment by d-torrance** [comment]
> a `NumericalImplicitization` test runs oom, don't think it's related to this PR?

Probably #3239

**Comment by mahrud** [comment]
I reran that test and it passed; LGTM!

**Pull request merged**